widgetssdk/src/main/java/com/glia/widgets/survey/SurveyController.java

```diff
@@ -104,6 +104,9 @@ public void onAnswer(@NonNull Survey.Answer answer) {
                             item.getAnswerCallback().answerCallback(showError);
                         }
                     }
+                    if (item.getQuestion().getType() != Survey.Question.QuestionType.TEXT) {
```

Comment:
Lot of different stuff and nesting happening in `public void onAnswer()`. Maybe you can refactor it a bit? For example move everything in `.ifPresent()` into a separate method.

Reply:
resolved